ndsl/dsl/stencil.py

```diff
@@ -797,6 +797,25 @@ def get_origin_domain(
             domain[i] += 2 * n
         return tuple(origin), tuple(domain)
 
+    def get_2d_compute_origin_domain(
+        self,
+        klevel: int = 0,
+    ) -> Tuple[Tuple[int, ...], Tuple[int, ...]]:
+        """
+        Get the origin and domain for a computation that occurs on the lowest klevel over a certain grid
+        configuration (given by dims) and a certain number of halo points.
+
+        Args:
+            klevel: the vertical level of the domain, defaults to zero
+
+        Returns:
+            origin: origin of the computation
+            domain: shape of the computation
+        """
+        origin = (self.isc, self.jsc, klevel)
```

Comment:
Maybe it's because I don't understand the application of the resulting stencil, but why do you need to send and return klevel as part of `origin`?